zilencer/views.py

```diff
@@ -592,7 +600,13 @@ def remote_server_post_analytics(
 
 
 def get_last_id_from_server(server: RemoteZulipServer, model: Any) -> int:
-    last_count = model.objects.filter(server=server).order_by("remote_id").only("remote_id").last()
+    last_count = (
+        model.objects.filter(server=server)
+        .exclude(remote_id=None)
```

Comment:
This tweak is needed because Postgres treats NULLs as the largest value, so we would always return `None`s from this function otherwise if such a row exists. See `test_get_last_id_from_server_ignores_null` added in this commit.

Reply:
This is great -- just posted a separate comment about putting this in a more separated commit for readability. And we should make sure we've got a comment explaining this exclusion; something like "# Rows with remote_id=None are managed by the bouncer service itself, not items to be synced.".